packages/numcodecs/meta.yaml

```diff
@@ -8,6 +8,10 @@ source:
     - patches/fixblosc.patch
     - patches/fixsetup.patch
     - patches/fixzlib.patch
+build:
+  script: |
+    rm numcodecs/blosc.c
```

Comment:
Yeah. `blosc.c` is the result of applying cython to `blosc.pyx`. We patch `blosc.pyx` in `patches/fixblosc.patch`. If Cython is installed, it will notice that `blosc.pyx` is new and recythonize it with the patched source, replacing `blosc.c`. If Cython is not installed, it will build with the old `blosc.c`. Removing `blosc.c` forces the build to fail when Cython is missing, which makes it less confusing.

On a related note, I am super confused about why the `fixblosc.patch` is needed at all anymore. Originally we needed it because `multiprocessing.Lock()` raises `ModuleNotFoundError: _multiprocessing not found`, whereas the `.pyx` code only caught `OSError`. (On a side side note, we should consider making `Lock` available as a no-op: since there is no multiprocessing, implementing locking by doing nothing is correct.)

Anyways numcodecs has since been patched to also catch `ImportError`:
https://github.com/zarr-developers/numcodecs/blame/v0.9.1/numcodecs/blosc.pyx#L82
I believe that `ModuleNotFoundError` is a subtype of `ImportError` so it should be covered. Indeed on native Python, the following import succeeds with no error:
```py
import multiprocessing
def patch_Lock():
	raise ModuleNotFoundError()
multiprocessing.Lock = patch_Lock
import numcodecs.blosc
```
However, the same code with the unpatched version of numcodecs in Pyodide fails with `ModuleNotFoundError`. I simply cannot comprehend why this happens. I spent a significant amount of time trying to locate the problem in the debugger before giving up.

Reply:
Thanks for the detailed explanation!

> On a side side note, we should consider making `Lock` available as a no-op: since there is no multiprocessing, implementing locking by doing nothing is correct.

Yes, we could consider monkey-patching some of multiprocessing functionalities.